pkgs/by-name/ed/edopro/package.nix

```diff
@@ -0,0 +1,320 @@
+
+      runHook postInstall
+    '';
+  };
+
+  edopro = stdenv.mkDerivation {
+    pname = "edopro";
+    version = deps.edopro-version;
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = "edopro";
+      rev = deps.edopro-rev;
+      hash = deps.edopro-hash;
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [
+      bzip2
+      curl
+      flac
+      fmt_9
+      freetype
+      irrlicht-edopro
+      libevent
+      libgit2
+      libjpeg
+      libpng
+      libvorbis
+      lua5_3
+      nlohmann_json
+      ocgcore
+      openal
+      SDL2
+      sqlite
+    ];
+
+    # nixpkgs' gcc stack currently appears to not support LTO
+    postPatch = ''
+      substituteInPlace premake5.lua \
+        --replace-fail 'flags "LinkTimeOptimization"' 'removeflags "LinkTimeOptimization"'
+    '';
+
+    preBuild = ''
+      premake5 gmake2 \
+        --architecture=${archLabel} \
+        --covers=\"${covers_url}\" \
+        --fields=\"${fields_url}\" \
+        --pics=\"${pics_url}\" \
+        --prebuilt-core="${lib.getLib ocgcore}/lib" \
+        --sound=sfml
+    '';
+
+    enableParallelBuilding = true;
+    env = {
+      # remove after release 40.1.4+
+      # https://discord.com/channels/170601678658076672/792223685112889344/1286043823293599785
+      CXXFLAGS = "-include cstdint";
+      LDFLAGS = "-I ${irrlicht-edopro}/include -L ${irrlicht-edopro}/bin";
+    };
+    buildFlags = [
+      "verbose=true"
+      "config=release_${archLabel}"
+      "ygopro"
+    ];
+    makeFlags = [
+      "-C"
+      "build"
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin
+      cp bin/${archLabel}/release/ygopro $out/bin
+
+      runHook postInstall
+    '';
+  };
+
+  edopro-script =
+    let
+      assetsToCopy = lib.concatStringsSep "," [
+        # Needed if we download files from ProjectIgnis' website or any https only website.
+        "cacert.pem"
+        "config"
+        "deck"
+        "COPYING.txt"
+        "expansions"
+        "lflists"
+        "notices"
+        "puzzles"
+        "fonts"
+        "script"
+        "skin"
+        "sound"
+        "textures"
+        "WindBot"
+      ];
+    in
+    writeShellScriptBin "edopro" ''
+      set -eu
+      EDOPRO_DIR="''${XDG_DATA_HOME:-$HOME/.local/share}/edopro"
+
+      if [ ! -d $EDOPRO_DIR ]; then
+          mkdir -p $EDOPRO_DIR
+          cp -r --no-preserve=all ${assets.src}/{${assetsToCopy}} $EDOPRO_DIR
+          chmod -R go-rwx $EDOPRO_DIR
+
+          rm $EDOPRO_DIR/config/io.github.edo9300.EDOPro.desktop.in
+      fi
+
+      export PATH=PATH:'${lib.makeBinPath [ mono ]}';
+      export LD_LIBRARY_PATH='${
+        lib.makeLibraryPath [
+          libGL
+          libX11
+          libxkbcommon
+          libXxf86vm
+          sqlite
+          wayland
+          egl-wayland
+        ]
+      }';
+
```

Comment:
I think it would be better if we moved this to the wrapping of `edopro`, and have this script only handle the user env setup & `-C` arg adding.

```suggestion
```

Reply:
Remember, the more you change, the more you have to maintain.